openrr-nav-viewer/examples/dwa_gui.rs

```diff
@@ -27,6 +27,37 @@ fn robot_path_from_vec_vec(path: Vec<Vec<f64>>) -> RobotPath {
     RobotPath(robot_path_inner)
 }
 
+fn linear_interpolate_path(path: Vec<Vec<f64>>, extend_length: f64) -> Vec<Vec<f64>> {
+    if path.len() < 2 {
+        return path;
+    }
+    let mut interpolated_path = vec![];
+    interpolated_path.push(path.first().unwrap().clone());
+    for i in 0..(path.len() - 1) {
```

Comment:
LGTM!!!

To put a finer point on it, but to make it short, it can be written like this.

```suggestion
    for (p0, p1) in path.iter().zip(path.iter().skip(1)) {
```

Reply:
It sounds great! I updated!